validation-test/stdlib/Collection/LazyMapCollection.swift

```diff
@@ -0,0 +1,175 @@
+// -*- swift -*-
+
+// RUN: rm -rf %t
+// RUN: mkdir -p %t
+// RUN: %S/../../../utils/gyb %s -o %t/main.swift
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-build-swift %t/main.swift -o %t/LazyMapCollection.swift.a.out
+// RUN: %S/../../../utils/line-directive %t/main.swift -- %target-run %t/LazyMapCollection.swift.a.out
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+// Also import modules which are used by StdlibUnittest internally. This
+// workaround is needed to link all required libraries in case we compile
+// StdlibUnittest with -sil-serialize-all.
+import SwiftPrivate
+#if _runtime(_ObjC)
+import ObjectiveC
+#endif
+
+var CollectionTests = TestSuite("Collection")
+
+// Test collection using value types as elements.
+CollectionTests.addRandomAccessCollectionTests (
+  makeCollection: { (elements: [OpaqueValue<Int>]) in
+    MinimalSequence(elements: elements).lazy.map(identity)
+  },
+  wrapValue: identity,
+  extractValue: identity,
+  makeCollectionOfEquatable: { (elements: [MinimalEquatableValue]) in
+    MinimalSequence(elements: elements).lazy.map(identityEq)
```

Comment:
Same comments here: does returning MinimalSequence in RandomAccessCollection tests compile?  And please gyb the repetition.